cuda/extract_cudart.rb

```diff
@@ -1,13 +1,52 @@
 require_relative 'extract_base.rb'
 
+begin
+  preprocessed_sources_libc = $cpp.preprocess(<<EOF).gsub(/^#.*?$/, '')
+#include <stdint.h>
+EOF
+rescue
+  C::Preprocessor.command = "gcc -E"
+  preprocessed_sources_libc = $cpp.preprocess(<<EOF).gsub(/^#.*?$/, '')
+#include <stdint.h>
+EOF
+end
+
+$parser.parse(preprocessed_sources_libc)
```

Comment:
Maybe:

```suggestion
begin
  preprocessed_sources_libc = $cpp.preprocess
#include <stdint.h>
EOF
rescue
  C::Preprocessor.command = "gcc -E"
  preprocessed_sources_libc = $cpp.preprocess
#include <stdint.h>
EOF
end

$parser.parse(preprocessed_sources_libc.).gsub(/^#.*?$/, ''))
```

Just to make clure that the gsub is not the problem, and the same in both case